core/arch/arm/plat-stm/platform_config.h

```diff
@@ -190,132 +177,37 @@
 #define SCU_CTRL_INIT			0x00000065
 
 /*
- * TEE RAM layout without CFG_WITH_PAGER:
- *
- *  +---------------------------------------+  <- CFG_DDR_TEETZ_RESERVED_START
- *  | TEE private secure |  TEE_RAM         |   ^
- *  |   external memory  +------------------+   |
- *  |                    |  TA_RAM          |   |
- *  +---------------------------------------+   | CFG_DDR_TEETZ_RESERVED_SIZE
- *  | Secure Data Path test memory (opt.)   |   |
- *  +---------------------------------------+   |
- *  |     Non secure     |  SHM             |   |
- *  |   shared memory    |                  |   |
- *  +---------------------------------------+   v
- *
- *  TEE_RAM : default 1MByte
- *  TA_RAM  : all what is left
- *  SDP_RAM : optional default SDP test memory 8MByte
- *  PUB_RAM : default 2MByte
- *
- * ----------------------------------------------------------------------------
- * TEE RAM layout with CFG_WITH_PAGER=y:
- *
- *  +---------------------------------------+  <- CFG_CORE_TZSRAM_EMUL_START
- *  | TEE private highly | TEE_RAM          |   ^
- *  |   secure memory    |                  |   | CFG_CORE_TZSRAM_EMUL_SIZE
- *  +---------------------------------------+   v
- *
- *  +---------------------------------------+  <- CFG_DDR_TEETZ_RESERVED_START
- *  | TEE private secure |  TA_RAM          |   ^
- *  |   external memory  |                  |   |
- *  +---------------------------------------+   | CFG_DDR_TEETZ_RESERVED_SIZE
- *  | Secure Data Path test memory (opt.)   |   |
- *  +---------------------------------------+   |
- *  |     Non secure     |  SHM             |   |
- *  |   shared memory    |                  |   |
- *  +---------------------------------------+   v
- *
- *  TEE_RAM : default 256kByte
- *  TA_RAM  : all what is left in DDR TEE reserved area
- *  SDP_RAM : optional default SDP test memory 8MByte
- *  PUB_RAM : default 2MByte
+ * Register nsecure ddr chunks for dynamic shm: these are DDR ranges
+ * that do not include OP-TEE secure memory.
+ * Some stm platforms may reserve begining of the DDR for non REE memory.
  */
 
-/* default locate shared memory at the end of the TEE reserved DDR */
-#ifdef CFG_SHMEM_SIZE
-#define TEE_SHMEM_SIZE		CFG_SHMEM_SIZE
-#else
-#define TEE_SHMEM_SIZE		(2 * 1024 * 1024)
-#endif
+#define MAX2(x, y)	(((x) > (y)) ? (x) : (y))
```

Comment:
Why do we need another set of `MIN/MAX`, what about the one in `utils.h`?

Reply:
yes, i could have change **util.h** to support `MIN`/`MAX` macros for assembly code.
Will do.